beszel/internal/agent/update.go

```diff
@@ -53,4 +54,100 @@ func Update() {
+
+	// Handle SELinux context if needed
+	handleSELinuxContext()
+
+	// Try to restart the service if it's running
+	restartService()
+}
+
+// restartService attempts to restart the beszel-agent service
+func restartService() {
+	// Check if we're running as a service by looking for systemd
+	if _, err := exec.LookPath("systemctl"); err == nil {
+		// Check if beszel-agent service exists and is active
+		cmd := exec.Command("systemctl", "is-active", "beszel-agent.service")
+		if err := cmd.Run(); err == nil {
+			fmt.Println("Restarting beszel-agent service...")
+			restartCmd := exec.Command("systemctl", "restart", "beszel-agent.service")
+			if err := restartCmd.Run(); err != nil {
+				fmt.Printf("Warning: Failed to restart service: %v\n", err)
+				fmt.Println("Please restart the service manually: sudo systemctl restart beszel-agent")
+			} else {
+				fmt.Println("Service restarted successfully")
+			}
+			return
+		}
+	}
+
+	// Check for OpenRC (Alpine Linux)
+	if _, err := exec.LookPath("rc-service"); err == nil {
+		cmd := exec.Command("rc-service", "beszel-agent", "status")
+		if err := cmd.Run(); err == nil {
+			fmt.Println("Restarting beszel-agent service...")
+			restartCmd := exec.Command("rc-service", "beszel-agent", "restart")
+			if err := restartCmd.Run(); err != nil {
+				fmt.Printf("Warning: Failed to restart service: %v\n", err)
+				fmt.Println("Please restart the service manually: sudo rc-service beszel-agent restart")
+			} else {
+				fmt.Println("Service restarted successfully")
+			}
+			return
+		}
+	}
+
+	// Check for OpenWRT procd
+	if _, err := exec.LookPath("service"); err == nil {
+		cmd := exec.Command("service", "beszel-agent", "running")
+		if err := cmd.Run(); err == nil {
+			fmt.Println("Restarting beszel-agent service...")
+			restartCmd := exec.Command("service", "beszel-agent", "restart")
+			if err := restartCmd.Run(); err != nil {
+				fmt.Printf("Warning: Failed to restart service: %v\n", err)
+				fmt.Println("Please restart the service manually: sudo service beszel-agent restart")
+			} else {
+				fmt.Println("Service restarted successfully")
+			}
+			return
+		}
+	}
+
+	fmt.Println("Note: Service restart not attempted. If running as a service, restart manually.")
+}
+
+// handleSELinuxContext applies SELinux context if SELinux is enabled
+func handleSELinuxContext() {
+	// Check if SELinux is enabled
+	cmd := exec.Command("getenforce")
+	output, err := cmd.Output()
+	if err != nil {
+		return // SELinux not available
+	}
+
+	if strings.TrimSpace(string(output)) == "Disabled" {
+		return // SELinux is disabled
+	}
+
+	fmt.Println("SELinux enabled, applying context...")
+
+	// Try chcon first
+	if chconCmd, err := exec.LookPath("chcon"); err == nil {
+		binaryPath, _ := os.Executable()
+		chconExec := exec.Command(chconCmd, "-t", "bin_t", binaryPath)
+		if err := chconExec.Run(); err != nil {
+			fmt.Println("Warning: chcon command failed to apply context.")
+		}
+	}
+
+	// Try restorecon as well
```

Comment:
@svenvg93 just wondering, but does this command not just revert the context back to the default?

> The `restorecon` command is used to restore the SELinux security context of files and directories to their default values as defined in the SELinux policy.